File: backend/core-api/src/main/java/io/basearchitecture/core/service/AiServiceHealthService.java
package io.basearchitecture.core.service;

import io.basearchitecture.core.client.ai.AiServiceClient;
import io.basearchitecture.core.client.ai.AiServiceClientException;
import io.basearchitecture.core.client.ai.AiServiceHealthPayload;
import org.springframework.stereotype.Service;

@Service
public class AiServiceHealthService {

    private static final String EXPECTED_STATUS = "up";
    private static final String EXPECTED_SERVICE = "base-architecture-ai-service";

    private final AiServiceClient aiServiceClient;

    public AiServiceHealthService(AiServiceClient aiServiceClient) {
        this.aiServiceClient = aiServiceClient;
    }

    public AiServiceHealthResult getHealth() {
        AiServiceHealthPayload payload;
        try {
            payload = aiServiceClient.getHealth();
        } catch (AiServiceClientException exception) {
            throw AiServiceHealthException.fromClient(exception);
        }

        if (payload == null
                || !EXPECTED_STATUS.equals(payload.status())
                || !EXPECTED_SERVICE.equals(payload.service())) {
            throw AiServiceHealthException.invalidContract();
        }

        return new AiServiceHealthResult(payload.status(), payload.service());
    }
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/_health_ai_service/service/AiServiceHealthService.kt
package ai.govbiz.core._health_ai_service.service

import ai.govbiz.core._common.exception.AiServiceCallException
import ai.govbiz.core._health_ai_service.client.AiServiceHealthClient
import ai.govbiz.core._health_ai_service.service.dto.AiServiceHealthResult
import org.springframework.stereotype.Service

@Service
class AiServiceHealthService(
    private val aiServiceHealthClient: AiServiceHealthClient,
) {
    fun getHealth(): AiServiceHealthResult {
        val payload = aiServiceHealthClient.getHealth()

        val status = payload.status
        val service = payload.service
        if (status != EXPECTED_STATUS || service != EXPECTED_SERVICE) {
            throw AiServiceCallException.invalidResponse(
                "AI Service health response violated the expected contract",
                null,
            )
        }

        return AiServiceHealthResult(status, service)
    }

    private companion object {
        const val EXPECTED_STATUS = "up"
        const val EXPECTED_SERVICE = "govbiz-ai-service"
    }
}
